src/remote_gym/remote_environment_management.py

```diff
@@ -222,64 +356,78 @@ def Process(self, request_iterator, context):
                 logging.debug(f"Received message of type {message_type}.")
 
                 if message_type == "create_world":
-                    environment_should_be_reset = True
                     response = dm_env_rpc_pb2.CreateWorldResponse(world_name="world")
 
+                    packed_args = internal_request.settings["args"]
+                    args = (
+                        {}
+                        if packed_args.WhichOneof("payload") is None
+                        else json.loads(tensor_utils.unpack_tensor(packed_args))
+                    )
+                    self.new_environment(context.peer(), args)
+
                 elif message_type == "join_world":
-                    environment_should_be_reset = True
+                    # Make sure to shutdown when the client leaves
+                    # todo not sure if this adds double callbacks
```

Comment:
What do you mean?

Reply:
Are you referencing here the fact that you add a callback and *additionally* call this on "destroy_world"?

Reply:
Theoretically one could join, leave, join, leave and thus add more and more callbacks. We don't do that, but its possible I believe.

Reply:
Irrelevant now, since you now only set the callback on `create_world`.